app/uk/gov/hmrc/servicecommissioningstatus/service/StatusCheckService.scala

```diff
@@ -32,56 +32,78 @@ class StatusCheckService @Inject()(
-                           .map(status => acc + (env -> status))
-                       ).map(AppConfigEnvironment.apply)
-
-      releases      <- releasesConnector.getReleases(serviceName).map(_.versions)
-      deploymentEnv  = DeploymentEnvironment(Environment.values.map(env => env -> checkIsDeployedForEnv(serviceName, releases, env)).toMap)
-
-      kibana      <- serviceConfigsConnector.getKibanaDashboard(serviceName)
-      grafana     <- serviceConfigsConnector.getGrafanaDashboard(serviceName)
-      buildJobs   <- serviceConfigsConnector.getBuildJobs(serviceName)
-      alertConfig <- serviceConfigsConnector.getAlertConfig(serviceName)
-    } yield ServiceCommissioningStatus(
-      serviceName       = serviceName
-    , hasRepo           = repo
-    , hasSMConfig       = smConfig
-    , hasFrontendRoutes = envRoutes
-    , hasAppConfigBase  = appConfigBase
-    , hasAppConfigEnv   = appConfigEnvs
-    , isDeployed        = deploymentEnv
-    , hasDashboards     = Dashboard(kibana, grafana)
-    , hasBuildJobs      = buildJobs
-    , hasAlerts         = alertConfig
-    )
-  }
-
-  private def checkRepoExists(serviceName: String)(implicit hc: HeaderCarrier): Future[StatusCheck] =
-    for {
-      exists <- OptionT(gitHubConnector.getGithubRaw(s"/hmrc/$serviceName/main/repository.yaml"))
-        .orElse(OptionT(gitHubConnector.getGithubApi(s"/repos/hmrc/$serviceName")))
-        .value
+      appConfigEnvs <- Environment
+                        .values
+                        .foldLeftM(Map.empty[Environment, Check.Result]) {
+                          (acc, env) => checkAppConfigExistsForEnv(serviceName, env).map(r => acc + (env -> r))
+                        }
+      isFrontend    <- isFrontend(serviceName)  // TODO check serviceType?
+      routes        <- if (isFrontend) serviceConfigsConnector.getMDTPFrontendRoutes(serviceName)
+                       else            Future.successful(Nil)
+      frontendRoutes = Environment
+                        .values
+                        .map(env => env -> checkFrontendRoute(routes, env)) // TODO make optional
+                        .toMap
+      buildJobs     <- serviceConfigsConnector.getBuildJobs(serviceName)
+      smConfig      <- checkServiceManagerConfigExists(serviceName)
+      deploymentEnv <- releasesConnector
+                        .getReleases(serviceName)
+                        .map(releases => Environment.values.map(env => env -> checkIsDeployedForEnv(serviceName, releases.versions, env)).toMap)
+      kibana        <- serviceConfigsConnector.getKibanaDashboard(serviceName)
+      grafana       <- serviceConfigsConnector.getGrafanaDashboard(serviceName)
+      alertConfig   <- serviceConfigsConnector.getAlertConfig(serviceName)
     } yield
-      if (exists.isDefined) StatusCheck(Some(s"https://github.com/hmrc/$serviceName"))
-      else StatusCheck(None)
+      SimpleCheck(title = "Github Repo"           , checkResult  = githubRepo     ) ::
```

Comment:
if wrapping the checkResult into a SimpleCheck/EnCheck is done on construction, there is no risk of mixing them up. the yield can just define the relative ordering of the checks.

Reply:
I found it useful to see all the titles in one place. There's always a chance of getting them wrong no matter how it's implemented. Still trying to work out what to do with all the private functions in this class. Will review this in future changes.